Android/Java/app/src/main/java/com/ironsource/ironsourcesdkdemo/DemoActivity.java

```diff
@@ -119,51 +134,54 @@ public void showRewardedVideoButtonTapped(View view) {
 
     public void loadInterstitialButtonTapped(View view) {
         // This will load an Interstitial ad
-
+        mInterstitialAd = new LevelPlayInterstitialAd(INTERSTITIAL_AD_UNIT_ID);
+        mInterstitialAd.setListener(new DemoInterstitialAdListener(this));
+        mInterstitialAd.loadAd();
         log("loadInterstitial");
-        IronSource.loadInterstitial();
     }
 
     public void showInterstitialButtonTapped(View view) {
         // It is advised to make sure there is available ad before attempting to show an ad
-        if (IronSource.isInterstitialReady()) {
+        if (mInterstitialAd.isAdReady()) {
             // This will present the Interstitial.
             // Unlike Rewarded Videos there are no placements.
 
             log("showInterstitial");
-            IronSource.showInterstitial();
+            mInterstitialAd.showAd(this);
         } else {
             // load a new ad before calling show
         }
     }
 
     public void loadBannerButtonTapped(View view) {
-        // call IronSource.destroyBanner() before loading a new banner
-        if (bannerParentLayout != null) {
-            this.destroyBanner();
+        // call destroy() before loading a new banner
+        if (mBannerAd != null) {
+            destroyBanner();
         }
 
         // choose banner size
-        // you can pick any banner size: ISBannerSize.BANNER, ISBannerSize.LARGE, ISBannerSize.RECTANGLE, ISBannerSize.SMART or even define a custom banner size by providing width and height
-        ISBannerSize size = ISBannerSize.BANNER;
+        // 1. recommended - Adaptive ad size that adjusts to the screen width
+        // 2. Adaptive ad size using fixed width ad size
+        // 3. Specific banner size - BANNER, LARGE, MEDIUM_RECTANGLE
+        LevelPlayAdSize adSize = LevelPlayAdSize.createAdaptiveAdSize(this);
 
-        // initialize IronSourceBanner object, using the IronSource.createBanner API
-        ironSourceBannerLayout = IronSource.createBanner(this, size);
+//        LevelPlayAdSize adSize = LevelPlayAdSize.createAdaptiveAdSize(this, 400);
+//        LevelPlayAdSize adSize = LevelPlayAdSize.BANNER;
 
-        if (ironSourceBannerLayout != null) {
-            // add IronSourceBanner to your container
-            FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams(MATCH_PARENT, MATCH_PARENT);
-            bannerParentLayout.addView(ironSourceBannerLayout, 0, layoutParams);
+        // Create the banner view and set the ad unit id and ad size
+        mBannerAd = new LevelPlayBannerAdView(this, BANNER_AD_UNIT_ID);
+        mBannerAd.setAdSize(adSize);
 
-            // set the banner listener
-            ironSourceBannerLayout.setLevelPlayBannerListener(new DemoBannerAdListener(this));
+        // add LevelPlayBannerAdView to your container
+        FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams(MATCH_PARENT, MATCH_PARENT);
+        bannerParentLayout.addView(mBannerAd, 0, layoutParams);
```

Comment:
I think it should be moved to "onCreate" function

Reply:
not sure which lines you mean, I moved some but not the lines for the frame layout, I don't think they should be part of the move